rewrite-yaml/src/main/java/org/openrewrite/yaml/MergeYamlVisitor.java

```diff
@@ -18,71 +18,98 @@
 import lombok.RequiredArgsConstructor;
 import org.openrewrite.Cursor;
 import org.openrewrite.ExecutionContext;
-import org.openrewrite.internal.ListUtils;
+import org.openrewrite.internal.lang.Nullable;
 import org.openrewrite.yaml.tree.Yaml;
 
+import java.util.ArrayList;
 import java.util.List;
 import java.util.concurrent.atomic.AtomicInteger;
 
 import static java.util.Objects.requireNonNull;
+import static org.openrewrite.internal.ListUtils.concatAll;
+import static org.openrewrite.internal.ListUtils.map;
+import static org.openrewrite.yaml.MappingUtils.keyMatches;
 
 @RequiredArgsConstructor
-public class MergeYamlVisitor extends YamlVisitor<ExecutionContext> {
+public class MergeYamlVisitor extends YamlIsoVisitor<ExecutionContext> {
+    @Nullable
     final Yaml scope;
```

Comment:
This shouldn't be `@Nullable`, right? It looks like the visit methods will all throw an exception if it is, so you might as well require a non-null value up front.

Reply:
The only time it will actually be null will be the first time it's invoked from the top level, when getVisitor is called. every other time it's instantiated i should have a scope.